ERCS/erc-7730.md

```diff
@@ -0,0 +1,1033 @@
+
+In this example, the metadata section contains only the recipient information, in the form of a displayable name (`owner` key) and additional information (`info` key) that MAY be used by wallets to provide details about the recipient.
+
+Finally, the `display` section contains the definitions of how to format each field of targeted contract calls under the `formats` key. 
+
+In this example, the function being described is identified by its solidity signature `transfer(address,uint256)`. This is th signature used to compute the function selector `0xa9059cbb` (using the solidity sgnature guarantees unicity in the conext of the contract). 
+* The `intent` key contains a human readable string that wallets SHOULD display to explain to the user the intent of the function call. 
+* The `fields` key contains all the parameters that CAN be displayed, and the way to format them 
+  * The `required` key indicates which parameters wallets SHOULD display). 
+  
+In this example, the `_to` parameter and the `_value` SHOULD both be displayed, one as an address replaceable by a trusted name (ENS or others), the other as an amount formatted using metadata of the target ERC-20 contract (USDT). 
+
+### Common concepts
+
+#### Key naming convention
+
+In all the specification, key names starting with `$` are *internal* and have no value beyond readability of the specification file itself. They should not be used in any function to build the UI to review structured data.
+
+#### Structured Data
+
+This specification intends to be extensible to describe the display formatting of any kind of *structured data*. 
+
+By *Structured data*, we target any data format that has:
+* A well defined *type* system; the data being described itself being of a specific type
+* A description of the type system, the *schema*, that should allow splitting the data into *fields*, each field clearly identified with a *path* that can be descrived as a string.
+
+Displaying structured data is often done by wallets to review its content before authorizing an action in the form of a *signature* over some serialization of the structured data. As such, the structured data is contained in a *container structure*:
+* Container structure has a well defined *signature* scheme (a serialization scheme, a hashing scheme, and signature algorithm).
+* The container structure does not necessarily follow the same type system as the structured data.
+* Wallets receive the full container structure and uses the signature scheme to generate a signature on the overall structure.
+
+```mermaid
+block-beta
+  columns 1
+  block:container
+    columns 3
+    title["Container Structure"] space:2
+    a["Data"]
+    struct["Structured Data"]
+    c["Data"]
+  end
+  blockArrowId4<["Wallet"]>(down)
+  block:signature
+    columns 3
+    space sig["Signature"] space
+  end
+  style title fill:transparent,stroke:transparent
+  style struct fill:#555
+```
```

Comment:
Mermaid is not rendered by our tooling. Please replace this with an image in your `assets/` directory. We prefer SVG, but PNG is also acceptable.